src/components/container/DailyDataChart/DailyDataChart.tsx

```diff
@@ -0,0 +1,270 @@
+        }
+    }
+
+    const GraphToolTip = ({ active, payload, label }: any) => {
+        if (active && payload && payload.length) {
+            var date = payload[0].payload.date;
+            return (
+                <div className="mdhui-daily-data-tooltip">
+                    <div className="mdhui-daily-data-tooltip-value">
+                        {props.valueFormatter ? props.valueFormatter(payload[0].payload.rawValue) : payload[0].payload.value}
+                    </div>
+                    <div className="mdhui-daily-data-tooltip-date">{format(date, 'MM/dd/yyyy')}</div>
+                </div>
+            );
+        }
+        return null;
+    };
+
+    const DayTick = ({ x, y, stroke, payload }: any) => {
+        var value = payload.value;
+        if (intervalType == "Month") {
+            let currentDate = new Date(intervalStart.getFullYear(), intervalStart.getMonth(), value);
+            return <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 15} textAnchor="middle" fontSize="12">{value}</text>;
+        } else {
+            let currentDate = intervalStart;
+            let dayOfWeek = format(currentDate, "E").substr(0, 1);
+            let month = currentDate.getMonth();
+            for (let i = 0; i < 7; i++) {
+                if (currentDate.getDate() == value) {
+                    month = currentDate.getMonth();
+                    dayOfWeek = format(currentDate, "E").substr(0, 2);
+                    break;
+                }
+                currentDate = add(currentDate, { days: 1 });
+            }
+            month++;
+            return <>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 8} textAnchor="middle" fontSize="11">{dayOfWeek}</text>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 24} textAnchor="middle" fontSize="12">{value}</text>
+            </>;
+        }
+    }
+
+    function tickFormatter(args: any) {
+        if (args >= 10000) {
+            return Number((args / 1000).toFixed(1)) + 'K';
+        } else {
+            return Number(args.toFixed(1));
+        }
+        return args;
+    }
+
+    //ensures that gradients are unique for each chart
+    let barGradientKey = `barGradient_${Math.random()}`;
+    let areaGradientKey = `areaGradient_${Math.random()}`;
+    function standardChartComponents() {
+        let domain: AxisDomain | undefined = undefined;
+        if (props.options) {
+            if (props.chartType == "Line") {
+                let domainMin = (props.options as LineChartOptions).domainMin;
+                if (domainMin == "Auto") {
+                    domain = ["auto", "auto"];
+                } else if (domainMin != undefined) {
+                    domain = [domainMin, "auto"];
+                }
+            }
+        }
+
+        return <>
+            <defs>
```

Comment:
Could these defs be moved down with their respective charts since they aren't really applicable globally.  One or the other or both is not relevant for every chart at the moment.

Reply:
Ah yep, not sure why I did it this way